### DISCUSSION TITLE
Using the pathname ( available through usePathname() hook ) in the /pages router

### DISCUSSION THREAD

**Discussion by JeelRajodiya** [vercel/next.js#62880] [App Router] [answered]
So I have a next.js application ( version 14.0.1) that uses `/pages` router. 
I was using `router.asPath` to get the pathname, but it was resulting in hydration errors ( Error: Hydration failed because the initial UI does not match what was rendered on the server. ) 

So when I replaced the `router.asPath` with pathname ( available through `usePathname()` ). and the error was gone. 
However, I am concerned that it might have some side effect because, in the next.js docs usePathname is only used with the `/app` router.

Is it okay to use pathname in the `/pages` router?

you can see [this comment](https://github.com/json-schema-org/website/issues/389#issuecomment-1976384784) and the [whole discussion](https://github.com/json-schema-org/website/issues/389) for more context

**Comment by jaysoo** [comment, marked as answer]
FYI, it is supported in both page and app router as mentioned in the issue here: https://github.com/vercel/next.js/issues/54242

**Reply by JeelRajodiya** [reply]
Thanks
